elyra/pipeline/kfp/kfp_authentication.py

```diff
@@ -0,0 +1,605 @@
+        if (1) kf_endpoint is unsecured (2) kf_endpoint does not
+        support static password authentication (3) the credentials are invalid
+
+        :param kf_endpoint: Kubeflow API endpoint to verify
+        :type kf_endpoint: str
+        :param runtime_config_name: Runtime configuration name where kf_endpoint is specified
+        :type runtime_config_name: str
+        :param username: Id to be used for authentication
+        :type username: str
+        :param password: Password to be used for authentication
+        :type password: str
+        :raises AuthenticationError: Authentication failed due to the specified error.
+        :return: A cookie value
+        """
+        with requests.Session() as s:
+
+            request_history = []
+
+            ################
+            # Determine if Endpoint is Secured
+            ################
+            resp = s.get(kf_endpoint, allow_redirects=True)
+            request_history.append({'request_url': kf_endpoint, 'response': resp})
+            if resp.status_code != HTTPStatus.OK:
+                raise AuthenticationError(f'Error detecting whether Kubeflow server at {kf_endpoint} is secured: '
+                                          f'HTTP status code {resp.status_code}'
+                                          f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                          provider=self._type,
+                                          request_history=request_history)
+
+            if len(resp.history) == 0:
+                # if we were NOT redirected, then the endpoint is UNSECURED
+                # treat this as an error.
+                raise AuthenticationError(f'The Kubeflow server at {kf_endpoint} is not secured using LDAP. '
+                                          f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                          provider=self._type,
+                                          request_history=request_history)
+
+            ################
+            # Get Dex Login URL
+            ################
+            redirect_url_obj = urlsplit(resp.url)
+
+            # if we are at `/auth?=xxxx` path, we need to select the
+            # static password auth type
+            if re.search(r"/auth$", redirect_url_obj.path):
+                redirect_url_obj = redirect_url_obj._replace(
+                    path=re.sub(r"/auth$", "/auth/local", redirect_url_obj.path)
+                )
+            else:
+                # verify that KF is secured by LDAP
+                m = re.search(r"/auth/([^/]*)/?", redirect_url_obj.path)
+                if m and m.group(1) != 'local':
+                    raise AuthenticationError(
+                        f'The Kubeflow server at {kf_endpoint} is secured using \'{m.group(1)}\'. '
+                        f'Update runtime configuration \'{runtime_config_name}\' and try again.',
```

Comment:
This error message should indicate you MIGHT be using `Dex - {m.group(1)}`, there is nothing dex-specific about having `/auth/{local,ldap}` in your path.

Reply:
Done.